starknet_py/net/account/account.py

```diff
@@ -254,18 +255,35 @@ async def sign_declare_transaction(
         self,
         compiled_contract: str,
         *,
+        compiled_class_hash: Optional[int] = None,
         max_fee: Optional[int] = None,
         auto_estimate: bool = False,
-    ) -> Declare:
-        contract_class = create_compiled_contract(compiled_contract=compiled_contract)
-        declare_tx = Declare(
-            contract_class=contract_class,
-            sender_address=self.address,
-            max_fee=0,
-            signature=[],
-            nonce=await self.get_nonce(),
-            version=self.supported_transaction_version,
-        )
+    ) -> Union[Declare, DeclareV2]:
+        if compiled_class_hash is not None:
```

Comment:
I think a more explicit check for the v2 Declare would be better.
It's not said anywhere that the compiled_class_hash has to be given for the tx to count as of version 2.

Reply:
From what I understand, it is impossible to deploy cairo0 contract through declare v2 and vice versa. So if compile_class_hash is provided (and it only exists for cairo1 contracts) it means v2 must be used.

Reply:
Right, it should be impossible. What I meant is that if a user would want to send a Declare v2 right now, and did not pass `compiled_class_hash` (maybe thinking it'd be calculated internally), the error would be some parse fail gibberish.

Reply:
`compiled_contract` will have `sierra_program` if it is `DeclareV2`, so we can check if it is present and if so throw an error when `compiled_class_hash` is not provided

Reply:
That sounds like a very good idea, let's do this.